Mage.Sets/src/mage/cards/l/LightningCoils.java

```diff
@@ -0,0 +1,135 @@
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.ElementalToken;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ *
+ * @author escplan9 - Derek Monturo
+ */
+public class LightningCoils extends CardImpl {
+    
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("a nontoken creature you control");
+    static {
+        filter.add(new ControllerPredicate(TargetController.YOU));
+        filter.add(Predicates.not(new TokenPredicate()));
+    }
+
+    public LightningCoils(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{3}");
+        
+        // Whenever a nontoken creature you control dies, put a charge counter on Lightning Coils.
+        this.addAbility(
+                new DiesCreatureTriggeredAbility(
+                        new AddCountersSourceEffect(CounterType.CHARGE.createInstance(), true), 
+                        false, filter));
+        
+        // At the beginning of your upkeep, if Lightning Coils has five or more charge counters on it, remove all of them from it 
+        // and put that many 3/1 red Elemental creature tokens with haste onto the battlefield. 
+        // Exile them at the beginning of the next end step.
+        this.addAbility(new BeginningOfUpkeepTriggeredAbility(new LightningCoilsEffect(), TargetController.YOU, false));
+    }
+
+    public LightningCoils(final LightningCoils card) {
+        super(card);
+    }
+
+    @Override
+    public LightningCoils copy() {
+        return new LightningCoils(this);
+    }
+}
+
+class LightningCoilsEffect extends OneShotEffect {
+
+    LightningCoilsEffect() {
+        super(Outcome.Benefit);
+        staticText = "if {this} has five or more charge counters on it, remove all of them from it and put that man 3/1 red Elemental creature tokens with haste onto the battlefield. Exile them at the beginning of the next end step.";
+    }
+
+    LightningCoilsEffect(final LightningCoilsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Permanent p = game.getPermanent(source.getSourceId());
+        if (p != null && controller != null) {
+            int counters = p.getCounters(game).getCount(CounterType.CHARGE);
+            if (counters >= 5) {
+                // remove all the counters and create that many tokens
+                p.removeCounters(CounterType.CHARGE.getName(), p.getCounters(game).getCount(CounterType.CHARGE), game);
+                CreateTokenEffect effect = new CreateTokenEffect(new ElementalToken("CON", 1, true), counters);
+                effect.apply(game, source);
+                
+                // exile those tokens at next end step
+                for (UUID tokenId : effect.getLastAddedTokenIds()) {
```

Comment:
I was thinking of figuring out a way to refactor out this 'for loop' since there are many cards which generate tokens and then exile them at the beginning of the next end step. This was just copy paste from Chandra Flamecaller for instance.